src/librustc_mir/borrow_check.rs

```diff
@@ -931,41 +978,107 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
     }
 
     /// Check the permissions for the given lvalue and read or write kind
-    fn check_access_permissions(&self, (lvalue, span): (&Lvalue<'tcx>, Span), kind: ReadOrWrite) {
+    ///
+    /// Returns true if an error is reported, false otherwise.
+    fn check_access_permissions(&self,
+                                (lvalue, span): (&Lvalue<'tcx>, Span),
+                                kind: ReadOrWrite,
+                                is_local_mutation_allowed: LocalMutationIsAllowed)
+                                -> bool {
+        debug!("check_access_permissions({:?}, {:?}, {:?})",
+               lvalue, kind, is_local_mutation_allowed);
+        let mut error_reported = false;
         match kind {
             Write(WriteKind::MutableBorrow(BorrowKind::Unique)) => {
                 if let Err(_lvalue_err) = self.is_unique(lvalue) {
-                    span_bug!(span, "&unique borrow for `{}` should not fail",
-                        self.describe_lvalue(lvalue));
+                    let item_msg = match self.describe_lvalue(lvalue) {
+                        Some(name) => format!("`{}`", name),
+                        None => "item".to_owned()
+                    };
+                    span_bug!(span, "&unique borrow for {} should not fail", item_msg);
```

Comment:
There's no reason to use pretty-printing in `delay_span_bug` - it can hide information and it just makes the code uglier. Just format the lvalue directly (`format!("Accessing {:?} with the kind `{:?}` shouldn't be possible", lvalue, kind)`).